operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/Dependent.java

```diff
@@ -1,6 +1,11 @@
 package io.javaoperatorsdk.operator.api.reconciler.dependent;
 
+import static io.javaoperatorsdk.operator.api.reconciler.Constants.EMPTY_STRING;
+
 public @interface Dependent {
 
+  @SuppressWarnings("rawtypes")
   Class<? extends DependentResource> type();
+
+  String name() default EMPTY_STRING;
```

Comment:
Shouldn't have a constant `DEFAULT_NAME` ? would be more desriptive then `EMPTY_STRING`.

Reply:
We're using `EMPTY_STRING` elsewhere already to denote the absence of user-provided value. This is nothing new.

Reply:
Maybe then we should have `NO_VALUE`. Empty string has extremeely generic semantics.

Reply:
Sure but it's been like this since the beginning 😅

Reply:
it seems now it the time with v3 :)

Reply:
Should be done in a separate PR, though, imo.

Reply:
https://github.com/java-operator-sdk/java-operator-sdk/issues/1090